composites/Plugin/SocialPreviews/Facebook/helpers/determineFacebookImageProperties.js

```diff
@@ -0,0 +1,165 @@
+export const SQUARE_WIDTH = 158;
+export const SQUARE_HEIGHT = 158;
+export const PORTRAIT_WIDTH = 158;
+export const PORTRAIT_HEIGHT = 236;
+export const LANDSCAPE_WIDTH = 500;
+export const LANDSCAPE_HEIGHT = 261;
+
+/**
+ * Determines the image display mode of the Facebook Image.
+ *
+ * @param {Object} dimensions The image's dimensions.
+ *
+ * @returns {string} The display mode of the image.
+ */
+export function determineFacebookImageMode( dimensions ) {
+	if ( dimensions.height > dimensions.width ) {
+		return "portrait";
+	}
+
+	if ( dimensions.height === dimensions.width ) {
+		return "square";
+	}
+
+	return "landscape";
+}
+
+/**
+ * Gets the dimensions of the uploaded image.
+ *
+ * @param {string} src The image source.
+ *
+ * @returns {Object} The image original image dimensions.
+ */
+export function getOriginalImageDimensions( src ) {
+	/* istanbul ignore next */
+	return new Promise( ( resolve, reject ) => {
+		const img = new Image();
+
+		img.onload = () => {
+			resolve( {
+				width: img.width,
+				height: img.height,
+			} );
+		};
+
+		img.onerror = reject;
+
+		img.src = src;
```

Comment:
Promises are not my _forte_, but I'm not sure what `img.src = src;` is meant to do.
Also, on the "faulty image" example, I get an `Uncaught (in promise)` error in the console.